ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseProxy.java

```diff
@@ -43,22 +72,32 @@ public PubsubMessage apply(PubsubMessage message) {
       // Prevent null pointer exception
       message = PubsubConstraints.ensureNonNull(message);
 
+      if (proxyMatcher == null) {
+        loadResourcesOnFirstMessage();
+      }
+
       // Copy attributes
       Map<String, String> attributes = new HashMap<>(message.getAttributeMap());
 
       final String xpp = attributes.get(Attribute.X_PIPELINE_PROXY);
       String xff = attributes.get(Attribute.X_FORWARDED_FOR);
 
-      // If any of the configured proxyIpAddresses is present, then there's a GCP load balancer IP
+      // If any of the configured proxy IPs or subnets are present, then there's a proxy IP
       // we will need to remove; see
       // https://bugzilla.mozilla.org/show_bug.cgi?id=1729069#c8
-      for (String proxyIpAddress : proxyIps.split(",")) {
-        if (xff != null && xff.contains(proxyIpAddress)) {
-          xff = Arrays.stream(xff.split("\\s*,\\s*")) //
-              .filter(StringUtils::isNotBlank) //
-              .filter(ip -> !proxyIpAddress.equals(ip)) //
-              .collect(Collectors.joining(","));
+      // and
+      // https://bugzilla.mozilla.org/show_bug.cgi?id=1789992
+      if (xff != null) {
+        final String[] xffArray = xff.split("\\s*,\\s*");
+        final List<String> filteredXffList = Arrays.stream(xffArray) //
+            .filter(StringUtils::isNotBlank) //
+            .filter(ip -> !proxyMatcher.apply(ip)) //
+            .collect(Collectors.toList());
+        if (xffArray.length != filteredXffList.size()) {
+          xff = String.join(",", filteredXffList);
           countProxyIpAddress.inc();
+          attributes.put("detected_proxy", "true");
+          attributes.put("proxy_list_versions", proxyMatcher.version);
```

Comment:
per https://bugzilla.mozilla.org/show_bug.cgi?id=1789992#c22:
> some amount of visibility beyond what was specified originally is desirable

@whd do these fields satisfy that for you?